Adapter/Adapter_LegacyContext.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+/**
+ * 2007-2015 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ *  @author 	PrestaShop SA <contact@prestashop.com>
+ *  @copyright  2007-2015 PrestaShop SA
+ *  @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *  International Registered Trademark & Property of PrestaShop SA
+ */
+use PrestaShop\PrestaShop\Core\Foundation\Exception\DevelopmentErrorException;
+
+/**
+ * This adapter will complete the new architecture Context with legacy values.
+ * A merge is done, but the legacy values will be transfered to the new Context
+ * during legacy refactoring.
+ */
+class Adapter_LegacyContext
+{
+    /**
+     * To be used only in Adapters. Should not been called by Core classes. Prefer to use Core\Business\context class,
+     * that will contains all you need in the Core architecture
+     * .
+     * @return Context The Legacy context, for Adapter use only.
+     */
+    public function getContext()
+    {
+        $legacyContext = \Context::getContext();
+        if (!isset($legacyContext->shop) ||
+            !isset($legacyContext->language) ||
+            !isset($legacyContext->link)
+        ) {
+            throw new DevelopmentErrorException('Legacy context is not set properly. Cannot use it to merge with Context structure.');
+        }
+        return $legacyContext;
+    }
+
+    /**
+     * Will merge the legacy Context values inside the new Context object.
+     *
+     * @param PrestaShop\PrestaShop\Core\Business\Context $newContext
+     */
+    public function mergeContextWithLegacy(PrestaShop\PrestaShop\Core\Business\Context &$newContext)
```

Comment:
Can't we shorten this type hinting to Context by adding as ```php use ``` statement at the begining of the file ?